backfill of run requests uses sql for RunRequest association

dimroc opened the pull request:
https://www.pivotaltracker.com/story/show/165125877

1. In migration1554131520, switched from gorm to pure sql, because it would erroneously insert
after update, perhaps due to misuse.

Resulting migration became leaner

```sql
#(/Users/dimroc/go/src/github.com/smartcontractkit/chainlink/store/migrations/migration1554131520/migrate.go:69)
INSERT INTO "run_requests" ("request_id","tx_hash","requester","created_at") VALUES (NULL,NULL,NULL,'2019-04-06 12:34:33')

# (/Users/dimroc/go/src/github.com/smartcontractkit/chainlink/store/migrations/migration1554131520/migrate.go:73)
UPDATE "job_runs" SET run_request_id = '7'  WHERE id = '50b26fc70f7640418b2c8ba22be4293f'
```

from the gorm generated:

```sql
#(/Users/dimroc/go/src/github.com/smartcontractkit/chainlink/store/migrations/migration1554131520/migrate.go:70)
INSERT INTO "run_requests" ("request_id","tx_hash","requester","created_at") VALUES (NULL,NULL,NULL,'2019-04-06 12:34:33')

#(/Users/dimroc/go/src/github.com/smartcontractkit/chainlink/store/migrations/migration1554131520/migrate.go:73)
UPDATE "run_requests" SET "request_id" = NULL, "tx_hash" = NULL, "requester" = NULL, "created_at" = '2019-04-06 12:23:02'  WHERE "run_requests"."id" = '2'
UPDATE "job_runs" SET "run_request_id" = '2'  WHERE "job_runs"."deleted_at" IS NULL AND "job_runs"."id" = 'bab27ecbd6204bc1a38005e8e77fee95'
SELECT * FROM "job_runs"  WHERE "job_runs"."deleted_at" IS NULL AND "job_runs"."id" = 'bab27ecbd6204bc1a38005e8e77fee95' ORDER BY "job_runs"."id" ASC LIMIT 1
UPDATE "run_requests" SET "request_id" = NULL, "tx_hash" = NULL, "requester" = NULL, "created_at" = '2019-04-06 12:23:02'  WHERE "run_requests"."id" = '2'

[2019-04-06 12:23:02]  UNIQUE constraint failed: job_runs.id

# (/Users/dimroc/go/src/github.com/smartcontractkit/chainlink/store/migrations/migration1554131520/migrate.go:73)
INSERT INTO "job_runs" ("id","job_spec_id","result_id","run_request_id","status","created_at","completed_at","updated_at","initiator_id","creation_height","observed_height","overrides_id","deleted_at") VALUES ('bab27ecbd6204bc1a38005e8e77fee95','','0','2','','2019-04-06 12:23:02',NULL,'2019-04-06 12:23:02','0',NULL,NULL,'0',NULL)
```

2. Add a test so future migrations run off of a real existing db from migration1554131520. Only supports sqlite, because of the challenge around seeding both sqlite and postgres. Doesn't test any current migrations, intended for future migrations. Attempted build flags, but wasn't better than `t.Skip(...)`: https://gist.github.com/dimroc/cf26e385e8f3e313935415f06cb57dc3

se3000 commented on store/migrations/migrate_test.go:
```diff
@@ -63,6 +65,33 @@ func TestMigrate_Upgrade(t *testing.T) {
+	}
+
+	require.NoError(t, os.MkdirAll(config.RootDir(), 0700))
+	cltest.WipePostgresDatabase(tc.Config)
```
```suggestion
	cltest.WipePostgresDatabase(config)
```

dimroc commented:
This PR has increased in complexity somewhat, with the following two points.

1. When updating the `store.BulkDeleteRuns` to clean up `RunRequests`, it seems that job runs aren't creating new run requests. (This line is inserting 0).

```
INSERT INTO "job_runs" ("id","job_spec_id","result_id","run_request_id","status","created_at","completed_at","updated_at","initiator_id","creation_height","observed_height","overrides_id","deleted_at") VALUES ('3b9320d54eb849479a5a21fb2c85e0c5','b8de668b1adc472a85515b27d4666965','7','0','completed','2019-04-08 07:23:28',NULL,'2019-04-08 07:23:28','1',NULL,NULL,'0',NULL)
```

2. I wanted to retrofit the `run_request_id` to be NOT NULL and a foreign key to `RunRequest`, since after backfilling w data, every job run should have a run request. Because of sqlite3's inability to ALTER TABLE and add foreign keys or do many other modifications, this results in a migration that has to create a new table, do a data migration, and then rename table.


Point 1 seems like a blocker, and would have been caught by 2.

dimroc commented:
This PR is ready for review again!

1. Updated `BulkDeleteRuns` to delete RunRequests
2. Investigated making `run_request_id` `NOT NULL` and a foreign key and ran into challenges outside the scope of the PR.

Challenges such as:
a) sqlite3 being unable to alter a column or a foreign key of a table that's already made.
b) attempts to recreate the `job_runs` table with the correct NOT NULL REFERENCES column cascading to `task_runs` because the existing `task_runs.job_run_id` has a foreign key to the original, renamed table.
c) lack of cascade delete working properly in BulkDeleteRuns leading to the need to set references to null before a subsequent delete 
d) variety of tests leaving run_request_id null.

References:
1. branch of attempt: https://github.com/smartcontractkit/chainlink/tree/bugs/fix-run-request-id-migration-SUPER
2. https://stackoverflow.com/questions/4897867/update-foreign-key-references-when-doing-the-sqlite-alter-table-trick
3. https://www.sqlite.org/lang_altertable.html
4. https://www.sqlite.org/foreignkeys.html
